utxo/src/cache.rs

```diff
@@ -179,9 +179,9 @@ impl<'a> UtxosCache<'a> {
 
         tx.outputs()
             .iter()
-            // burned outputs should not be included into utxo set
-            .filter(|output| !output.purpose().is_burn())
             .enumerate()
+            // burned outputs should not be included into utxo set
+            .filter(|(_, output)| !output.purpose().is_burn())
```

Comment:
I don't think this makes a difference. You can put it back up there. It's even more readable.

Reply:
It actually does. I introduced a dedicated test for that. And it fails if I revert this line

Reply:
Actually you're right. We shouldn't filter before we enumerate. Never mind.